shaders/src/depth_main.fs

```diff
@@ -13,4 +13,15 @@ void main() {
         discard;
     }
 #endif
+
+#if defined(HAS_VSM)
+    // Since we're rendering from the perspective of the light, frameUniforms.cameraPosition is the
+    // light position, in world space.
+    highp float depth = length(frameUniforms.cameraPosition.xyz - vertex_worldPosition);
```

Comment:
I'm confused by this code. It is calculating the distance from the light to the fragment and it's calling that the "depth". Is that different from the actual depth?

Reply:
By "actual depth", do you mean depth in clip space (`gl_FragDepth`)? We _could_ use `gl_FragDepth`, but we need a linear depth representation. `gl_FragDepth` is only linear in specific cases (non-LiSPM directional lights with orthographic projections).

Reply:
You can easily linearize the depth, it's basically just a `madd`, this is done in `sao.mat`

But what I'm not sure about is that "the distance between the light and the fragment" is not the same than the "difference between the z of the camera and the fragment's" -- so which is the correct one?
So conceptually, what is it that we want? the depth or the distance to the camera?

Reply:
Ah I see the confusion. "distance from the fragment to the light" is the metric I'm going for.

> A linear metric with more uniform precision, such as the distance to the light, is more appropriate and stable.

[GPU Gems source](https://developer.nvidia.com/gpugems/gpugems3/part-ii-light-and-shadows/chapter-8-summed-area-variance-shadow-maps)